cdap-api/src/main/java/co/cask/cdap/api/security/securestore/SecureStoreMetadata.java

```diff
@@ -0,0 +1,184 @@
+import java.util.Map;
+
+/**
+ * Represents the metadata for the data stored in the Secure Store.
+ */
+public final class SecureStoreMetadata {
+
+  private static final String NAME_FIELD = "name";
+  private static final String DESCRIPTION_FIELD = "description";
+  private static final String CREATED_FIELD = "created";
+  private static final String PROPERTIES_FIELD = "properties";
+  private static final String DESCRIPTION_DEFAULT = "";
+  private static final Gson GSON = new Gson();
+
+  private final String name;
+  private final String description;
+  private final Date created;
+  private final Map<String, String> properties;
+
+  private SecureStoreMetadata(String name, String description, Date created, Map<String, String> properties) {
+    this.name = name;
+    this.description = description;
+    this.created = created;
+    this.properties = properties;
+  }
+
+  public static SecureStoreMetadata of(String name, Map<String, String> properties) {
+    String tempDescription = properties.get(DESCRIPTION_FIELD) == null ?
+      DESCRIPTION_DEFAULT : properties.get(DESCRIPTION_FIELD);
+
+    return new SecureStoreMetadata(name, tempDescription, new Date(), properties);
+  }
+
+  /**
+   * @return Name of the data.
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * @return Last time, in epoch, this element was modified.
+   */
+  public long getLastModifiedTime() {
+    return created.getTime();
+  }
+
+  /**
+   * @return A map of properties associated with this element.
+   */
+  public Map<String, String> getProperties() {
+    return properties;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public Date getCreated() {
+    return created;
+  }
+
+  @Override
+  public String toString() {
+    return "SecureStoreMetadata{" +
+      "name='" + name + '\'' +
+      ", description='" + description + '\'' +
+      ", created=" + created +
+      ", properties=" + properties +
+      '}';
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    SecureStoreMetadata that = (SecureStoreMetadata) o;
+    if (!name.equals(that.name)) {
+      return false;
+    }
+    if (!description.equals(that.description)) {
+      return false;
+    }
+    if (!created.equals(that.created)) {
+      return false;
+    }
+    return properties != null ? properties.equals(that.properties) : that.properties == null;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = name.hashCode();
+    result = 31 * result + description.hashCode();
+    result = 31 * result + created.hashCode();
+    result = 31 * result + (properties != null ? properties.hashCode() : 0);
+    return result;
+  }
+
+  /**
+   * Serialize the metadata to a byte array.
+   *
+   * @return the serialized bytes
+   * @throws IOException
+   */
+  public byte[] serialize() throws IOException {
```

Comment:
Better move this method out of this class. Keep this class a simple POJO and use Gson at the place where serialization/deserialization is actually needed (e.g. in the key store implementation).